async.md

```diff
@@ -218,3 +220,100 @@ Example
 			resp.error("caught: "+reason.message);
 		})
 ~~~
+
+# Messaging
+
+This class allows for the execution of async ClearBlade messaging functions provide an MQTT client that will work asynchronously. You can connect to the internal ClearBlade broker or an external broker. 
+
+## MQTT.Message(payload)
+
+The message represents an MQTT message.
+
+* @param {string} payload
+* @constructor
+* @property {string} payload
+* @property {number} qos
+* @property {boolean} retain
+* @property {boolean} duplicate
+
+Example
+
+~~~javascript
+	var msg = new MQTT.Message()
+ 	var msgData = JSON.parse(msg.payload);
+~~~
+
+## MQTT.Client(options)
+
+The client represents an MQTT client connected to a broker.
+If no options are specified, the client connects to the ClearBlade broker.
+
+* @param {object} [options]
+* @param {string} options.address
+* @param {number} [options.port=1883] 
+* @param {string} [options.username] 
+* @param {string} [options.password]
+* @param {string} [options.client_id={randomUUID}]
+* @param {boolean} [options.use_tls=false]
+* @param {object} [options.tls_config]
+* @param {string} options.tls_config.client_cert
+* @param {string} options.tls_config.client_key
+* @param {string} options.tls_config.ca_cert
+* @constructor
+
+Example
+
+~~~javascript
+var client = new MQTT.Client();
+var options = {
+  address: 'platform.clearblade.com',
```

Comment:
Not sure if using one of our brokers is the best example here, because I don't want users to think they need to use options all the time and specify the platform url.